lib/Db/ObjectHandlers/MariaDbFacetHandler.php

```diff
@@ -0,0 +1,863 @@
+                    )
+                )
+            );
+        }
+
+        // Apply metadata filters from @self
+        if (isset($baseQuery['@self']) && is_array($baseQuery['@self'])) {
+            foreach ($baseQuery['@self'] as $field => $value) {
+                if ($value === 'IS NOT NULL') {
+                    $queryBuilder->andWhere($queryBuilder->expr()->isNotNull($field));
+                } else if ($value === 'IS NULL') {
+                    $queryBuilder->andWhere($queryBuilder->expr()->isNull($field));
+                } else if (is_array($value)) {
+                    $queryBuilder->andWhere($queryBuilder->expr()->in($field, $queryBuilder->createNamedParameter($value, \Doctrine\DBAL\Connection::PARAM_STR_ARRAY)));
+                } else {
+                    $queryBuilder->andWhere($queryBuilder->expr()->eq($field, $queryBuilder->createNamedParameter($value)));
+                }
+            }
+        }
+
+        // Apply JSON object field filters (non-@self filters)
+        $objectFilters = array_filter($baseQuery, function($key) {
+            return $key !== '@self' && !str_starts_with($key, '_');
+        }, ARRAY_FILTER_USE_KEY);
+
+        foreach ($objectFilters as $field => $value) {
+            $jsonPath = '$.' . $field;
+            
+            if ($value === 'IS NOT NULL') {
+                $queryBuilder->andWhere($queryBuilder->expr()->isNotNull(
+                    $queryBuilder->createFunction("JSON_EXTRACT(object, " . $queryBuilder->createNamedParameter($jsonPath) . ")")
+                ));
+            } else if ($value === 'IS NULL') {
+                $queryBuilder->andWhere($queryBuilder->expr()->isNull(
+                    $queryBuilder->createFunction("JSON_EXTRACT(object, " . $queryBuilder->createNamedParameter($jsonPath) . ")")
+                ));
+            } else if (is_array($value)) {
+                // For array values, check if any of the values match
+                $orConditions = $queryBuilder->expr()->orX();
+                foreach ($value as $val) {
+                    $orConditions->add(
+                        $queryBuilder->expr()->eq(
+                            $queryBuilder->createFunction("JSON_UNQUOTE(JSON_EXTRACT(object, " . $queryBuilder->createNamedParameter($jsonPath) . "))"),
+                            $queryBuilder->createNamedParameter($val)
+                        )
+                    );
+                }
+                $queryBuilder->andWhere($orConditions);
+            } else {
+                $queryBuilder->andWhere(
+                    $queryBuilder->expr()->eq(
+                        $queryBuilder->createFunction("JSON_UNQUOTE(JSON_EXTRACT(object, " . $queryBuilder->createNamedParameter($jsonPath) . "))"),
+                        $queryBuilder->createNamedParameter($value)
+                    )
+                );
+            }
+        }
+
+    }//end applyBaseFilters()
```

Comment:
This should be in an database-specific service/handler eventually, as JSON_EXTRACT is mariadb/mysql specific.

Furthermore, we now only check equals, which breaks the functionality of fetching an object by a value in an array.